flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/allocator/DefaultSlotAssigner.java

```diff
@@ -48,14 +50,30 @@ public Collection<SlotAssignment> assignSlots(
             allGroups.addAll(createExecutionSlotSharingGroups(vertexParallelism, slotSharingGroup));
         }
 
-        Iterator<? extends SlotInfo> iterator = freeSlots.iterator();
+        Iterator<? extends SlotInfo> iterator =
+                selectSlotsInMinimalTaskExecutors(freeSlots, allGroups, Collections.emptyList())
+                        .iterator();
```

Comment:
I feel the current change is a little complex. IIUC, `DefaultSlotAssigner` only needs to change one line is enough.

All changes of `SlotAssigner` are not needed. They could simplified into `StateLocalitySlotAssigner`.

Please correct me if my understanding is wrong.

```suggestion
        Iterator<? extends SlotInfo> iterator = freeSlots.stream().sorted(Comparator.comparing(SlotInfo::getTaskManagerLocation)).iterator();
```